app/helpers/application_helper.rb

```diff
@@ -1042,6 +1042,25 @@ def perfmenu_click?
   end
 
   def javascript_redirect(args)
+    if args.class == Hash && args[:action] == 'show' && restful_routed?(self.class.model)
+      args.delete(:action)
+      polymorphic_path_redirect(model, args)
+    else
+      non_polymorphic_path_redirect(args)
+    end
+  end
+
+  def polymorphic_path_redirect(model, args)
+    args[:record] ? record = args[:record] : record = model.find(args[:id] || params[:id])
+    args.delete(:record)
+    args.delete(:id)
+    render :update do |page|
```

Comment:
In the other refactorings (and in the next steps also in refactoring of `javascript_redirect` we are getting rid of RJS. That means that all the `render :update do` blocks have to go.

This way you are actually adding a new one.

Can you, please, change this so that we do not introduce any new such blocks to the codebase?